src/screens/NotificationSettings/index.tsx

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019-2020 @Premiurly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import { Button, Form, Switch } from 'antd';
+import React, { useEffect, useState } from 'react';
+import { useUserDetailsContext } from 'src/context';
+import queueNotification from 'src/ui-components/QueueNotification';
+
+import { useChangeNotificationPreferenceMutation } from '../../generated/graphql';
+import { handleTokenChange } from '../../services/auth.service';
+import { NotificationStatus } from '../../types';
+import FilteredError from '../../ui-components/FilteredError';
+
+const NotificationSettings = () => {
+	const currentUser = useUserDetailsContext();
+	const [changed, setChanged] = useState<boolean>(false);
+	const [postParticipated, setPostParticipated] = useState<boolean>(false);
+	const [postCreated, setPostCreated] = useState<boolean>(false);
+	const [newProposal, setNewProposal] = useState<boolean>(false);
+	const [ownProposal, setOwnProposal] = useState<boolean>(false);
+	const [changeNotificationPreferenceMutation, { error, loading }] = useChangeNotificationPreferenceMutation();
+	const { notification, email_verified } = currentUser;
+
+	useEffect(() => {
+		setPostParticipated(notification?.postParticipated || false);
+		setPostCreated(notification?.postCreated || false);
+		setNewProposal(notification?.newProposal || false);
+		setOwnProposal(notification?.ownProposal || false);
+	}, [notification]);
+
+	const updatePreference = () => {
+		changeNotificationPreferenceMutation({
+			variables: {
+				newProposal,
+				ownProposal,
+				postCreated,
+				postParticipated
+			}
+		})
+			.then(({ data }) => {
+				if (data && data.changeNotificationPreference && data.changeNotificationPreference.message) {
+					queueNotification({
+						header: 'Success!',
+						message: data.changeNotificationPreference.message,
+						status: NotificationStatus.SUCCESS
+					});
+					setChanged(false);
+				}
+				if (data?.changeNotificationPreference?.token) {
+					handleTokenChange(data.changeNotificationPreference.token, currentUser);
+				}
+			}).catch((e) => {
+				queueNotification({
+					header: 'Failed!',
+					message: e.message,
+					status: NotificationStatus.ERROR
+				});
+				console.error('change name error', e);
+			});
+	};
+	return (
+		<Form onFinish={() => updatePreference()} className='w-full bg-white shadow-md p-8 rounded-md flex flex-col gap-y-8'>
+			<header>
+				<h3
+					className='font-medium text-lg tracking-wide text-sidebarBlue'
+				>
+						Notification Settings
+				</h3>
+				<h5
+					className='font-normal text-sm text-navBlue mt-2'
+				>
+						Update your notification settings. You will be receiving your notifications on <span className='text-sidebarBlue'>hellopolkassembly@gmail.com</span>
```

Comment:
The email should not be hardcoded, but should show the user's email. This component should only be visible when a user has an email, else should show a link to add email.